kvdb-memorydb/CHANGELOG.md

```diff
@@ -5,3 +5,9 @@ The format is based on [Keep a Changelog].
 [Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
+### Fixed
+- `iter_from_prefix` behaviour synced with the `kvdb-rocksdb`
```

Comment:
let's wait a bit till we resolve this in https://github.com/paritytech/parity-common/pull/277#issuecomment-561634775

Reply:
Revert just this change & update added tests or put the whole PR on ice & wait?

Reply:
let's put the PR on ice for now, I'll try to resolve it soon though, sorry for the inconvenience!

Reply:
That's fine, it was expected. I'll address remaining comments and will put it on ice.

Reply:
P.S. While resolving this, I'd recommend to change method names.

* `iter_from_prefix` is fine for the old  `-memorydb` behavior
* `iter_from_prefix` is kind of misleading for the current `-rocksdb` behavior, `iter_with_prefix` sounds better

Like:

* `from` - everything from the first prefix occurrence (`skip_while`)
* `with` - everything matching the prefix (`filter`)

Even both can be implemented if there's an use case for them.

Reply:
cc @dvdplm
it seems we have decided on the new behavior as the docs agrees with it
https://github.com/paritytech/parity-common/blob/2c26418244653e4a0134f1dfaf5a6079fa5589cf/kvdb/src/lib.rs#L141
 
I would vote for a `prefix_iter`, but `iter_with_prefix` sounds better than `iter_from_prefix`, but let's open a separate PR for that

Reply:
Now we're bikeshedding but to me `iter_from_prefix` says "this is an iterator created using a prefix"; in other words the "from" is about how it was created and not so much what it contains (or doesn't). That said, I'm fine changing names too.

Reply:
```rust
/// Iterate over flushed data for a given column, starting from a given prefix.
```

In other words, if I understand it correctly, next steps:

* rollback `-memorydb` behavior back to `skip_while` in this PR
* modify the `-rocksdb` behavior - align with docs (`skip_while`) & `-memorydb` in another PR to keep these (unrelated) changes separated

Right?

Reply:
A bit puzzled now, since the docs says _starting from a given prefix_ which implies the old behavior to me, but the PR was approved, which means no rollback. Shouldn't the docstring be updated to something like `starting with a given prefix` then?

Reply:
haha, we're still understanding this sentense differently
Correct me if I'm wrong @dvdplm, we have decided to *not* rollback the changes, but instead rename the method in a separate PR (the name is a subject to bikeshedd)

Reply:
@ordian correct (@zrzka we discussed this over chat, which was rude of us, apologies).

The gist is essentially: "RocksDB iterators are weird and we prefer to give our users a more Rust-like iterator; the old behaviour of returning values with  `key >= prefix` is surprising and all consuming code we know of works around this with a `filter` so lets make that the default".

Reply:
> haha, we're still understanding this sentense differently

Okay, let it be then :) Resolving this conversation :)